src/webui/www/private/scripts/misc.js

```diff
@@ -275,6 +276,37 @@ window.qBittorrent.Misc ??= (() => {
         });
     };
 
+    const downloadFile = async (url, defaultFileName, errorMessage = undefined) => {
+        errorMessage ??= "QBT_TR(Unable to download file)QBT_TR[CONTEXT=HttpServer]";
+        try {
+            const response = await fetch(url);
+            if ((response.status >= 200) && (response.status < 300)) {
+                const blob = await response.blob();
+                const fileNamePrefix = "filename=";
+                let fileName = defaultFileName;
+                for (const part of (response.headers.get("content-disposition") ?? "").split(";").map(s => s.trim())) {
```

Comment:
Should we care if the file name contains semicolons (`;`) ?
If not, then I suppose you can also use regex to match the filename. The code could be shorter.

Reply:
Yes, good callout. Given that the header always has a prefix of `attachment; filename=`, it seems even easier to just strip that out.